File: rewrite-java-ecj/src/test/java/org/openrewrite/java/JavaEcjParserTest.java
package org.openrewrite.java;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.io.TempDir;
import org.openrewrite.InMemoryExecutionContext;
import org.openrewrite.java.tree.J;
import org.openrewrite.test.RewriteTest;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class JavaEcjParserTest implements RewriteTest {

    @Test
    void parseJavaSourceFromFile(@TempDir Path tempDir) throws Exception {
        File source = new File(tempDir.toFile(), "A.java");
        String initialSource = """
                    import java.io.Serializable;
                    public class A implements Serializable {
                        class B {
                        }
                    }
                """;
        Files.writeString(source.toPath(), initialSource);

        List<J.CompilationUnit> sources = JavaEcjParser.builder().build().parse(List.of(source.toPath()), tempDir, new InMemoryExecutionContext());

        assertThat(sources.get(0).printAll()).isEqualTo(initialSource);
    }
}

File: rewrite-java-ecj/src/test/kotlin/org/openrewrite/java/JavaEcjParserTest.kt
package org.openrewrite.java

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.openrewrite.Parser
import java.nio.file.Paths

class JavaEcjParserTest {
    @Test
    fun projectPath() {
        val inputs = listOf(
                Parser.Input(Paths.get("p/src/main/java/org/openrewrite/A.java")) { "".byteInputStream() },
                Parser.Input(Paths.get("p/src/main/java/io/moderne/B.java")) { "".byteInputStream() },
        )

        val projectPath = JavaEcjParser.projectPath(inputs)

        assertThat(projectPath.toPortableString()).isEqualTo("p/src/main/java")
    }
}
